source/module_basis/module_nao/two_center_table.cpp

```diff
@@ -222,8 +222,10 @@ void TwoCenterTable::_tabulate(const NumericalRadial* it1, const NumericalRadial
     // than two spherical Bessel transforms. By doing so, we achieve a good
     // consistency between the table and its derivative during interpolation.
     using ModuleBase::CubicSpline;
-    CubicSpline::build(nr_, rgrid_, table_.inner_most_ptr<double>(itab), dtable_.inner_most_ptr<double>(itab),
-            CubicSpline::BoundaryCondition::first_deriv, CubicSpline::BoundaryCondition::first_deriv,
-            0.0, 0.0);
+    CubicSpline::build(
+            nr_, rgrid_, table_.inner_most_ptr<double>(itab),
+            {CubicSpline::BoundaryType::first_deriv, 0.0},
```

Comment:
> what is the meaning of braces here ("{...}")?

This PR changes the way to specify a cubic spline boundary condition. This brace now serves as a list-initialization of the new "BoundaryCondition" class object. In the specific example, it means "a boundary condition of type first_deriv with value 0.0".

The construction of a cubic spline needs 2 boundary conditions, which could be:
- periodic
- not-a-knot
- a given first derivative
- a given second derivative

The question is that, periodic/not-a-knot boundary condition do not involve any user-specified values, while first/second deriv do. That is to say, not-a-knot alone is a well-defined boundary condition, while first-deriv is not; the latter must be accompanied by a value.

The previous version of CubicSpline has such values as the last two function arguments anyway (they are defaulted to 0.0 so users do not need to specify them when using not-a-knot or periodic):
```
CubicSpline(int n, double* x, double* y, BoundaryType start, BoundaryType end, double start=0.0, double end=0.0)
```
However, I felt such design was not very intuitive and can be sometimes confusing. Consider a mixed boudary condition with not-a-knot at start and first derivative at end. The old one would be
```
double deriv_val = 1.0;
CubicSpline cubspl(n, x, y, not_a_knot, first_deriv, 0.0, deriv_val); // 0.0 is not meaningful
```
not-a-knot is supposed to have no accompanied value, but in the old design one has to provide a value anyway. I think this could be confusing and error-prone.

To clarify the boundary condition, in this PR I chose to use a struct "BoundaryCondition" to specify an actual boundary condition:
```
    struct BoundaryCondition
    {
        // for not_a_knot and periodic
        BoundaryCondition(BoundaryType type = BoundaryType::not_a_knot);

        // for first/second_deriv
        BoundaryCondition(BoundaryType type, double val);

        BoundaryType type;
        double val = 0.0;
    };
```
This struct is simple enough that it merely holds an enum class & a value; the type and value are now group together for first/second_deriv. For the same example, the new code would be
```
double deriv_val = 1.0;
CubicSpline cubspl(n, x, y, not_a_knot, {first_deriv, deriv_val});
```